### PR TITLE
Extend NetworkEndpoint with more fields to support workload oriented policy enforcement

### PR DESCRIPTION
**Pull request by lei-tang**
Extend NetworkEndpoint with more fields to support workload oriented policy enforcement:
the namespace of the workload, the service account of the workload, and the labels
of the workload.

### CODE REVIEW

**Comment by rshriram** [comment]
what is workload oriented policy enforcement? is this a new thing? Aren't we already doing this today?

**Comment by lei-tang** [comment]
@rshriram The design document for workload oriented policy is at [this link](https://docs.google.com/document/d/1eInPqAjOfsyJWtMIOeRRza0b0aErKqVzO9mbxNKcucA/edit#heading=h.7zgnj8bwqfld).

**Comment by lei-tang** [comment]
#9123

**Comment by lei-tang** [comment]
https://github.com/istio/istio/issues/8990

**Comment by diemtvu** [comment]
@rshriram we enforce authN policy on workload, but the way to specify target is using service name. Following this PR, we will also change the policy to use workload instead (e.g, using label). This is why we need namespace & label information in endpoint so that we can find the right policy for it. Hope this clear.

File: pilot/pkg/model/service.go [hunk 1]
```diff
@@ -291,6 +291,16 @@ type NetworkEndpoint struct {
 	// catalog.mystore.com)
 	ServicePort *Port
 
+	// Namespace for the workload on this network endpoint.
+	// For the platforms without namespace, it will be empty.
+	Namespace string
```
Comment (diemtvu) [inline pilot/pkg/model/service.go]: I think you can combine all the changes to populate these new fields in on PR.

File: pilot/pkg/model/service.go [hunk 2]
```diff
@@ -291,6 +291,16 @@ type NetworkEndpoint struct {
...
+
+	// Service account for the workload on this network endpoint.
+	ServiceAccount string
+
+	// Labels for the workload on this network endpoint.
+	Labels Labels
```
Comment (diemtvu) [inline pilot/pkg/model/service.go]: All these can be also under new struct, say Attributes or EndpointAttributes (similar to Service)

Comment (lei-tang) [reply pilot/pkg/model/service.go]: Done.

File: pilot/pkg/model/service.go (hunk 1 above)
Comment (yangminzhu) [reply pilot/pkg/model/service.go]: Just wondering should these be further wrapped in a `WorkloadAttributes`?

It's the namespace (serviceaccount, labels) of the workload (pod) but without reading the comment, it looks like it's the namespace (serviceaccounts, labels) of the Endpoint itself.

Also note in `ServiceAttributes`, the `Name`, `Namespace` and `UID` are all referring to the service itself.

Comment (lei-tang) [reply pilot/pkg/model/service.go]: These attributes have been organized into a struct called EndpointAttributes.

Comment (lei-tang) [reply pilot/pkg/model/service.go]: Added the code to populate the EndpointAttributes in InstancesByPort(). 
Added a test TestServiceEndpointsAttributes() to test the EndpointAttributes of service instances.

**Pull request closed without merging**